client/jetpack-onboarding/controller.js

```diff
@@ -20,3 +23,19 @@ export const onboarding = ( context, next ) => {
 	context.primary = <JetpackOnboardingMain stepName={ context.params.stepName } />;
 	next();
 };
+
+export const siteSelection = ( context, next ) => {
+	const state = context.store.getState();
+
+	// TODO: move to a separate selector
+	const siteId = findKey( state.jetpackOnboarding.credentials, credentials => {
```

Comment:
My old pet peeve -- querying state in controllers is a bit unreliable, since state may not yet contain relevant information.

If we're deviating from the "usual" `siteSelection` paradigm anyway, let's go a step further. Let's totally axe the `concept of a "currently selected site" stored in Redux state; hence, let's also drop the `jetpackOnboardingSiteId` subtree (reducer, actions, selectors) introduced by this PR. I don't think it adds a lot of value; instead, let's just use the site slug fragment from the route, and use a selector to query the corresponding site ID from the `jetpackOnboarding` subtree (is that what we called it) in `Main`'s `connect()` and propagate it to individual steps (or just connect each individual step, if `Wizard` doesn't support that).